Gems/ROS2ScriptIntegration/Code/Source/Clients/ROS2ScriptIntegrationSystemComponent.h

```diff
@@ -0,0 +1,92 @@
+
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
+        ROS2ScriptIntegrationSystemComponent();
+
+        ~ROS2ScriptIntegrationSystemComponent();
+
+    protected:
+        // AZ::TickBus::Handler overrides ...
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        // ROS2ScriptIntegrationRequestBus::Handler overrides ...
+        void PublishStdMsgsString(const AZStd::string& topicName, const AZStd::string& value) override;
+        void PublishStdMsgEmpty(const AZStd::string& topicName) override;
+        void PublishStdMsgUInt32(const AZStd::string& topicName, const uint32_t value) override;
+        void PublishStdMsgInt32(const AZStd::string& topicName, const int32_t value) override;
+        void PublishStdMsgFloat32(const AZStd::string& topicName, const float value) override;
+        void PublishStdMsgBool(const AZStd::string& topicName, const bool value) override;
+        void PublishGeometryMsgsTwist(const AZStd::string& topicName, const AZ::Vector3& linear, const AZ::Vector3& angular) override;
+        void PublishGeometryMsgTransform(const AZStd::string& topicName, const AZ::Transform& transform) override;
+        void PublishGeometryMsgVector3(const AZStd::string& topicName, const AZ::Vector3& vector) override;
+        void PublishGeometryMsgQuaternion(const AZStd::string& topicName, const AZ::Quaternion& quaternion) override;
+        void PublishGeometryMsgPoint32(const AZStd::string& topicName, const AZ::Vector3& point) override;
+        void PublishGeometryMsgPoseStamped(
+            const AZStd::string& topicName, const AZStd::string& frame, const AZ::Transform& transform) override;
+
+        // SubscriberRequestBus overrides ...
+        void SubscribeToStdMsgBool(const AZStd::string& topicName) override;
+        void SubscribeToSensorMsgJoy(const AZStd::string& topicName) override;
+        void SubscribeToGeometryMsgPoseStamped(const AZStd::string& topicName) override;
+        void SubscribeToString(const AZStd::string& topicName) override;
+        void SubscribeToFloat32(const AZStd::string& topicName) override;
+        void SubscribeToUInt32(const AZStd::string& topicName) override;
+        void SubscribeToInt32(const AZStd::string& topicName) override;
+
+        // AZ::Component overrides ...
+        void Init() override;
+
+        void Activate() override;
+
+        void Deactivate() override;
+
+    private:
+        AZStd::mutex m_publisherMapMutex;
+        AZStd::unordered_map<AZStd::string, std::shared_ptr<rclcpp::PublisherBase>> m_publishers;
+
+        AZStd::mutex m_subscribersMapMutex;
+        AZStd::unordered_map<AZStd::string, std::shared_ptr<rclcpp::SubscriptionBase>> m_subscribers;
```

Comment:
Those maps do not support different publisher types on the same topic.
As far as i know ros2 does not forbid it.